src/tcgwars/logic/impl/gen4/MysteriousTreasures.groovy

```diff
@@ -3187,7 +3187,7 @@ public enum MysteriousTreasures implements LogicCardInfo {
             if (chosenCard)
               chosenCard.showToOpponent("Chosen card").moveTo(my.hand)
 
-            shuffleDeck()
+            if (choice == 1 && my.deck) shuffleDeck()
```

Comment:
Why can't this just be in the initial check?

Reply:
Could be optimized into happening before revealing the chosen card & moving it into your hand. Went 1:1 with IRL logic there.

The second `my.deck` check would still be used inside the initial conditional, so don't see much point in changing this.